### ISSUE TITLE
Q: Should we add code coverage to all repos?

### ISSUE THREAD

**Issue by wilzbach** [dlang-community/discussions#15]
Imho it's very useful to see at a PR whether the touched lines are actually tested.

**Comment by BBasile** [comment]
The problem is that i see much codecov problems when i look at Phobos pull requests, so i'm against.

**Comment by wilzbach** [comment]
> The problem is that i see much codecov problems when i look at Phobos pull requests, so i'm against.

Yes there were many problems and we had to eat quite some dogfood (also with the CodeCov guys).
However, since DConf17 we have completely removed the non-deterministic changes from Phobos, so everything since shows only the related changes. Btw Phobos is a really bad example, because it used to have a lot of random, non-seeded unittest and many concurrent tests. The projects here are single-threaded and (should be) deterministic.

See also: https://github.com/dlang/phobos/pull/5202 and https://github.com/dlang/phobos/pull/5375

**Comment by BBasile** [comment]
My concern is the red crosses. :x: is repulsive imo.

**Comment by wilzbach** [comment]
> My concern is the red crosses. :x: is repulsive imo.

Understood. I have made two PRs to start an experiment:

https://github.com/dlang-community/libdparse/pull/144
https://github.com/dlang-community/D-YAML/pull/67

What do you say about testing it out at one or two repos and then making a decision / further move?
If it really is buggy / shows red crosses, we can easily remove it.

**Comment by ZombineDev** [comment]
I am in favor of adding code coverage. Personally as a contributer or user of open-source projects I see code coverage as signal that the maintainers care about the quality of the project. I really can't see this as a way to repulse potential contributors.

**Comment by ZombineDev** [comment]
> What do you say about testing it out at one or two repos and then making a decision / further move? If it really is buggy / shows red crosses, we can easily remove it.

+1

**Comment by wilzbach** [comment]
I recently found out that CodeCov supports `informational: true` which will result in it being always green.

```
coverage:
  precision: 3
  round: down
  range: "80...100"

  # Learn more at https://docs.codecov.io/docs/commit-status
  status:
    project: off
    patch:
      default:
        informational: true
    changes: off
```

There's also the possibility to use a very high threshold.

See also: https://github.com/dlang/dmd/blob/master/.codecov.yml
